kubeflow/pipelines.py

```diff
@@ -74,6 +74,7 @@ def train_op(
     num_gpu,
     gpu_type,
     checkpoint_file=None,
+    kfp_log_dir="/",
```

Comment:
We should allow users to change this log directory from kubeflow pipeline instead of using hard-coded constants.

Reply:
Why do users need to change the log dir?

Reply:
I think we should not bother users with another argument of log dir at least in kubeflow pipeline. During local run they will have the option of changing log dir using CLI.

Reply:
We should leave it as option and set defaults.

Reply:
We don't need to espose this option to kubeflow pipeline UI, but this cannot be hard coded to create implicit knowledge between kubeflow pipeline container op and datasetinsights CLI.

Reply:
```
train = dsl.ContainerOp(
        name="train",
        image=docker,
        command=command,
        arguments=arguments,
        pvolumes={DATA_PATH: volume},
        file_outputs={"mlpipeline-ui-metadata": "/mlpipeline-ui-metadata.json"},
    )	  
```
The `file_outputs` is hard coded path which should be avoided. I think it should be derived from `kfp_log_dir` and `kfp_filename`. Also, we are missing metrics filenames.

Reply:
I'll add metrics for evaluate pipeline. Do we need it for train as well?

Reply:
@adason Apparently `file_outputs` can only have string in it. It's giving me compilation error if I'm using it as follows:

```
file_outputs={
            "mlpipeline-metrics": os.path.join(
                str(kfp_log_dir), str(kfp_metrics_filename)
            ),
            "mlpipeline-ui-metadata": os.path.join(
                str(kfp_log_dir), str(kfp_ui_metadata_filename)
            ),
        },
```

Getting this error: 

```
RuntimeError: Internal compiler error: Found unresolved PipelineParam.
Please create a new issue at https://github.com/kubeflow/pipelines/issues attaching the pipeline code and the pipeline package.
```

Reply:
`kfp_log_dir` and `kfp_metrics_filename` does not have to be kubeflow pipeline parameter, but we should be able to configure them in dataset insights cli instead of having these to be hard coded constants in datasetinsights python code.

Reply:
But we have to specify the location of the JSON before hand for building the pipeline yaml file. If we can't use pipeline param then how would we populate it on run time?

Reply:
Similar to how `train_data` variable is specified.

Reply:
But `train_data` is `DATA_PATH` which is a global variable. Do you mean we should use global variable to define `kfp_log_dir` and `kfp_metrics_filename`? That would still be hard coding ?